### DISCUSSION TITLE
I have notice that in my server cpu usage is high due to php-fpm

### DISCUSSION THREAD

**Discussion by IamParantapa1** [EasyEngine/easyengine#1754] [Support Requests] [answered]
Currently, i am running a 4 core 8 gb ram digital ocean server where i am running only one website. But after the last update 4.4.0, my server CPU usage got crazy high in htop notice that php-fpm is consuming lots of CPU resources. Currently, I am running 4.6.1. 
is there any solution to that?

![Screenshot 2022-12-12 141516](https://user-images.githubusercontent.com/42549482/207000918-d7aeab93-731e-4108-951f-4c03c972272f.png)

**Comment by IamParantapa1** [comment]
Please any one help

**Reply by NecronMt** [reply]
same here since upgrade

**Reply by IamParantapa1** [reply]
also, i have found one more issue check your docker, it is also creating so many Jason logs it is creating rediculous amount of file size. 

sudo find /var/lib/docker/containers/ -name "*.log" -exec ls -sha {} \; | sort -hk1 -r | head -1

once you run this command you can find the size. for my case it 31 gb. 

https://community.easyengine.io/t/excessive-disk-usage-ee-v-4-0-12/12712

issue is the same but the corn job is not working any solution for this issue

**Reply by mrrobot47** [reply]
In the last few updates, only the official PHP-FPM base image was updated to the latest one: [diff1](https://github.com/EasyEngine/dockerfiles/pull/145/files#diff-3056f1116a5fd6fca517485791ecb6dadd1f345e3c0f830e984b709e5141865d) [diff2](https://github.com/EasyEngine/dockerfiles/pull/145/files#diff-63cda043733d6fcc2108aeefaaa9fc99243abd61a227fe72a71a2a4c59305b0c) [diff3](https://github.com/EasyEngine/dockerfiles/commit/06344522ffef1a920548969f9a9c137baf1226db), no other changes have been done which could hamper the performance. Probably some issue with the latest official image? I will be checking on this in the coming days.

**Reply by mrrobot47** [reply]
Re: Logs https://github.com/EasyEngine/easyengine/discussions/1761#discussioncomment-4508841

**Reply by IamParantapa1** [reply]
Also, please look at MySQL or mariadb usage is also high for some reason what happened in there?
And as I mention the thread docker is generating to many log file please fix that too.

**Comment by mrrobot47** [comment, marked as answer]
Please update to the latest release `EE v4.6.4`, PHP performance issue has been fixed in it.
Update using `ee cli update`.

**Reply by NecronMt** [reply]
seems like the PHAR is broken. I tried updating but it doesn't work. Thanks for the fix, what was it?

**Reply by mrrobot47** [reply]
Please share error log.

**Reply by mrrobot47** [reply]
Please ensure that you have ~5GB free space on server before updating EasyEngine.

**Reply by mrrobot47** [reply]
Fix: Issue with newrelic agent, ref: https://github.com/EasyEngine/dockerfiles/pull/150#issue-1516220895
This was causing issues in the PHP image regardless of whether newrelic was in use by any site.

**Reply by NecronMt** [reply]
I do have 20GB of free space

`Updating EasyEngine to new version. This might take some time.

                                                                                                                         
         PHP Fatal error:  Uncaught Error: Call to undefined function EE\Site\Type\curl_init() in phar:///tmp/ee_63b2fb19259c18.40972898.phar/vendor/easyengine/site-command/src/helper/class-ee-site.php:1584
         Stack trace:                                                                                                    
         #0 phar:///tmp/ee_63b2fb19259c18.40972898.phar/vendor/easyengine/site-command/src/helper/class-ee-site.php(1375): EE\Site\Type\EE_Site_Command->check_www_or_non_www_domain()
         #1 phar:///tmp/ee_63b2fb19259c18.40972898.phar/vendor/easyengine/site-command/src/helper/class-ee-site.php(1036): EE\Site\Type\EE_Site_Command->www_ssl_wrapper()
         #2 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Migration/SiteContainers.php(115): EE\Site\Type\EE_Site_Command->enable()
         #3 [internal function]: EE\Migration\SiteContainers::enable_default_containers()                                
         #4 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/RevertableStepProcessor.php(56): call_user_func_array()   
         #5 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Migration/Containers.php(64): EE\RevertableStepProcessor->execute()
         #6 [internal function]: EE\Migration\Containers::start_container_migration()                                    
         #7 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/RevertableStepProcessor.php(56): call_user_func_array()   
         #8 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Runner.php(133): EE\RevertableStepProcessor->execute()    
         #9 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Runner.php(906): EE\Runner->migrate()                     
         #10 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Runner.php(899): EE\Runner->trigger_migration()          
         #11 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Runner.php(80): EE\Runner->maybe_trigger_migration()     
         #12 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Runner.php(721): EE\Runner->init_ee()                    
         #13 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/EE/Bootstrap/LaunchRunner.php(23): EE\Runner->start()       
         #14 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/bootstrap.php(74): EE\Bootstrap\LaunchRunner->process()     
         #15 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/init-ee.php(27): EE\bootstrap()                             
         #16 phar:///tmp/ee_63b2fb19259c18.40972898.phar/php/boot-phar.php(8): include('...')                            
         #17 /tmp/ee_63b2fb19259c18.40972898.phar(5): include('...')                                                     
         #18 {main}                                                                                                      
           thrown in phar:///tmp/ee_63b2fb19259c18.40972898.phar/vendor/easyengine/site-command/src/helper/class-ee-site.php on line 1584
         Warning: An Error occurred. Initiating clean-up.                                                                
         Warning: Exiting gracefully after rolling back. This may take some time.                                        
                                                                                                                         
                                                                                                                         
`

**Reply by mrrobot47** [reply]
Which PHP version are you using on server? Seems like it is missing `curl` extension. Please update the PHP cli on server to have the extension.

**Reply by NecronMt** [reply]
oh yes actually upon investigation that's what I noticed. Thank you

**Comment by NecronMt** [comment]
![image](https://user-images.githubusercontent.com/33217809/210450923-7733c0d5-e4ab-4639-b747-4d46d297c53f.png)

seems like I have quite a lot on memory from MariaDB, anything I should look into please?

**Reply by NecronMt** [reply]
Any news please? I still have high usage :/
